Dto

MartinKonupcik opened the pull request:
Basic things in Dto, introducing classes.

MartinKonupcik committed:
Add EvidencePojisteniDto project and related classes

Introduces a new project "EvidencePojisteniDto" to the solution, including a project file targeting .NET Framework 4.7.2. Adds new classes: EditInsuranceDto, InsuranceListDto, and UpdateInsuranceDto with basic structure. Configures MSBuild settings and includes an AssemblyInfo.cs file for assembly metadata.

MartinKonupcik committed:
Add project references and enhance DTOs and entities

Updated project files to include references for API and DTO projects.
Enhanced data structures in DTOs by adding new properties.
Introduced new classes for Contract, Person, and Policy with relevant properties and enumerations to better represent the application's domain model.

MartinKonupcik committed:
Remove EvidencePojisteni project references

The `EvidencePojisteni` project has been completely removed as a dependency from both `EvidencePojisteni.API.csproj` and `EvidencePojisteni.Web.csproj`. All references to the `EvidencePojisteni` project have also been eliminated from the solution file (`EvidencePojisteni.sln`), including its project configuration settings.

MartinKonupcik committed:
Refactor DTOs: Remove insurance classes, add contract items

Removed `EditInsuranceDto`, `InsuranceListDto`, and `UpdateInsuranceDto` classes. Added new DTOs: `EditContractDto`, `EditPersonDto`, `EditPolicyDto`, `ItemListDto`, `ListItemContractDto`, `ListItemPersonDto`, `ListItemPolicyDto`, `UpdateContractDto`, `UpdatePersonDto`, and `UpdatePolicyDto`. Updated `EvidencePojisteniDto.csproj` to reflect these changes, focusing on contracts and items.

MartinKonupcik committed:
Enhance DTOs with new properties and enumerations

- Added `Active` property to `EditContractDto` and `UpdateContractDto`.
- Expanded `EditPersonDto` and `UpdatePersonDto` with `FirstName`, `LastName`, `Phone`, and `Age`.
- Introduced `Type` enumeration in `EditPolicyDto` and `UpdatePolicyDto`.
- Enhanced `ListItemContractDto` and `ListItemPersonDto` with additional properties.

MartinKonupcik committed:
#

Add Id property and Type enum to ListItemPolicyDto

This commit introduces a new `Id` property of type `Guid`
and a `Type` enumeration to the `ListItemPolicyDto` class.
The `Type` enum includes values for `Life`, `Health`,
`Property`, `Vehicle`, and `Travel`. Additionally,
unnecessary `using` directives have been removed for
code cleanup.

Darkvikis commented on EvidencePojisteniDto/EditContractDto.cs:
```diff
@@ -0,0 +1,16 @@
+namespace EvidencePojisteniDto
+{
+    public class EditContractDto
+    {
```
You can edit less then you have in ListItem?

Darkvikis commented on EvidencePojisteniDto/ItemListDto.cs:
```diff
@@ -0,0 +1,17 @@
+
+namespace EvidencePojisteniDto
+{
+    public class ItemListDto
```
Whats this for?

Darkvikis commented on EvidencePojisteniDto/EditPolicyDto.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public class EditPolicyDto
+    {
+        public enum Type
```
Enums should be always in its own file and folder too.

Darkvikis commented on EvidencePojisteniDto/EditPolicyDto.cs:
```diff
@@ -0,0 +1,21 @@
+            Vehicle,
+            Travel
+        }
+        public string Name { get; set; }
```
You cant edit more then Name on Policy?

Darkvikis commented on EvidencePojisteniDto/UpdatePolicyDto.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public class UpdatePolicyDto
+    {
+        public enum Type
```
Create file PolicyType, where the enum will be an then in Dtos 
public PolicyType Type { get; set;}

Darkvikis commented on EvidencePojisteniDto/ListItemContractDto.cs:
```diff
@@ -0,0 +1,19 @@
+{
+    public class ListItemContractDto
+    {
+        public Guid ContractId { get; set; }
```
Do you show all of these in the List?

Darkvikis commented on EvidencePojisteniDto/ListItemPersonDto.cs:
```diff
@@ -0,0 +1,17 @@
+namespace EvidencePojisteniDto
+{
+    public class ListItemPersonDto
+    {
```
Do you show all of these in the List?

Darkvikis commented on EvidencePojisteniDto/ListItemPolicyDto.cs:
```diff
@@ -0,0 +1,18 @@
+namespace EvidencePojisteniDto
+{
+    public class ListItemPolicyDto
+    {
```
Do you show all of these in the List?

MartinKonupcik committed:
Refactor DTOs and introduce PolicyType class

- Added ContractId, PersonId, and PolicyId to EditContractDto.
- Replaced Type enum in EditPolicyDto with PolicyType.Type and added Id property.
- Removed ItemListDto and updated project file to include PolicyType.cs.
- Removed PolicyId from ListItemContractDto and Phone, Age from ListItemPersonDto.
- Updated ListItemPolicyDto and UpdatePolicyDto to use PolicyType.Type instead of Type enum.
- Created new PolicyType class with an enum for policy types.

MartinKonupcik committed:
Remove unnecessary properties from EditContractDto

The properties `ContractId`, `PersonId`, and `PolicyId`
were removed from the `EditContractDto` class in the
`EvidencePojisteniDto` namespace to streamline the data model.

MartinKonupcik committed:
Refactor ContractController to use DTOs for responses

Updated ContractController to return ListItemContractDto instead of Contract. Modified New and Update methods to accept EditContractDto and UpdateContractDto, respectively. Changed PersonController's GetList method to return ListItemContractDto[]. Added new DTO classes for better data encapsulation and maintainability. Updated using directives in both controllers to include new DTOs.

MartinKonupcik committed:
Refactor person creation and update methods

Updated `New` method in `PersonController` to use `EditPersonDto` and return `IActionResult`. Modified `Update` method to accept `UpdatePersonDto` and check for ID mismatch. Added `PersonId` property to `UpdatePersonDto` for identification.

MartinKonupcik committed:
Refactor policy management to use DTOs and enums

Updated `PolicyController` to utilize DTOs for policy operations, changing method signatures and return types. Replaced the `Type` enum in `Policy` with a `PolicyType` enum for better structure. Adjusted `PolicyService` and related DTOs to align with these changes. Enhanced `PolicyType` to include a broader range of policy types. Overall, these modifications improve the clarity and maintainability of the policy management system.

MartinKonupcik committed:
Refactor API controllers and DTOs for clarity

- Updated `ContractController` to streamline property assignments and improve `GetList` method.
- Enhanced `PersonController` with route attributes, updated return types, and improved response handling.
- Similar updates made to `PolicyController` for consistency and clarity.
- Modified DTOs to remove default values and ensure consistent property definitions.
- Added additional properties to `ListItemPersonDto` and `ListItemPolicyDto` for better data representation.
- Overall improvements enhance the clarity, consistency, and functionality of the API.

MartinKonupcik committed:
Refactor contract management with DTOs

- Updated `ContractController` to use `EditContractDto` for new contract creation.
- Added `NotFound` response in `PolicyController` for better error handling.
- Refactored `ContractService`:
  - Changed contract list type to `List<ListItemContractDto>`.
  - Updated methods to use `contractId` and new DTOs for data handling.
- Introduced `EditContractDto` and `ListItemContractDto` for contract data management.
- Modified `UpdateContractDto` to include `ContractId` for accurate updates.
- Overall restructuring for improved data handling and separation of concerns.

Darkvikis commented on EvidencePojisteni.API/Entities/Policy.cs:
```diff
@@ -0,0 +1,11 @@
+    {
+        public Guid Id { get; set; }
+        public PolicyType Type { get; set; }
+        public string Name { get; set; } = null!;
```
add
MaxAmount
ValidFrom
ValidTo

Darkvikis commented on EvidencePojisteni.API/Controllers/PolicyController.cs:
```diff
@@ -1,89 +1,98 @@
+    [HttpPost]
+    public async Task<IActionResult> New([FromBody] EditPolicyDto policyDto)
+    {
+        var policy = new Policy
```
We talked about this. It shoulndt be here.
Only Dtos are in Controllers
find all of them

Darkvikis commented on EvidencePojisteni.API/Controllers/ContractController.cs:
```diff
@@ -44,10 +63,20 @@ public async Task<ActionResult<Contract[]>> GetList()
     /// </returns>
     [HttpPost]
-    public async Task New([FromBody] Contract contract)
+    public async Task<IActionResult> New([FromBody] EditContractDto contractDto)
```
Rename to Create and all Creates should return DetailDto

Darkvikis commented on EvidencePojisteni.API/Controllers/ContractController.cs:
```diff
@@ -56,31 +85,40 @@ public async Task New([FromBody] Contract contract)
+        };
+
+        var updated = await service.Update(contract);
+        if (!updated)
```
should have brackets

MartinKonupcik committed:
Refactor DTOs and entity structure in controllers

Updated `ContractController` and `PersonController` to use new DTOs (`NewContractDto`, `DetailContractDto`, `NewPersonDto`, `DetailPersonDto`). Refactored `Contract`, `Person`, and `Policy` classes to inherit from a new base class `Entity`, introducing a standardized `Id` property. Removed old DTO classes and updated the project file to reflect these changes. New DTO classes have been defined to align with the updated entity structure.

MartinKonupcik committed:
Refactor entities and DTOs for improved structure

- Updated `Contract` class in `Contract.cs` with new properties.
- Modified `Person` class in `Person.cs` to include `ContractId` and `PolicyId`, added constructor and update method, and introduced `GetListItem` method.
- Renamed `UpdatePersonDto` to `DetailPersonDto` for clarity.
- Renamed `UpdatePolicyDto` to `DetailPolicyDto` to reflect focus on details.
- Cleaned up `ListItemPersonDto` by removing unnecessary namespaces and added `ContractId` and `PolicyId` properties.

MartinKonupcik committed:
Add ContractId and PersonId to Policy and ListItemPolicyDto

Updated the `Policy` class to include `ContractId` and `PersonId` properties. Added constructors for initialization and an `Update` method for modifying `Type` and `Name`. Modified `GetListItem` to return a new `ListItemPolicyDto` with the new properties.

Updated `ListItemPolicyDto` to include `ContractId` and `PersonId` to align with changes in the `Policy` class.

MartinKonupcik committed:
Refactor contract handling in controller and service

Updated `ContractController` to return contract objects directly, simplifying data handling. Changed `New` method to accept `DetailContractDto` and added null checks. Cleaned up `Update` method for readability.

In `ContractService`, switched from `ContractId` to `Id` for consistency across methods. Modified `Create` to initialize contracts with `PolicyId` and `PersonId` from the DTO. Updated `Get` and `Delete` methods to reflect property name changes.

MartinKonupcik committed:
Refactor DTOs in Person and Contract services

Updated PersonController to use ListItemPersonDto and DetailPersonDto. Simplified methods to utilize GetListItem() for DTO creation. Modified ContractService to change property name from ContractId to Id. Changed PersonService to store ListItemPersonDto instead of Person objects, and introduced DetailPersonDto with PersonId. Enhanced ListItemPersonDto to include PersonId, ContractId, and PolicyId.

MartinKonupcik committed:
Refactor controllers and service for DTO handling

Updated `PersonController` and `PolicyController` to streamline service interactions and improve DTO management. The `New`, `Delete`, and `Update` methods have been simplified for better clarity and efficiency. Transitioned `PolicyService` to utilize `ListItemPolicyDto` instead of `Policy` objects, with new properties added to support functionality. Cleaned up unnecessary `using` directives in `DetailPolicyDto.cs`.

MartinKonupcik committed:
Fix route key in New method and improve Update readability

Updated the return statement in the New method to use 'Id' instead of 'PolicyId' in the route values.
Reformatted the Update method's if statement to include braces for better readability and adherence to coding standards.

MartinKonupcik committed:
Update insurance management system views and models

Refactor several files to enhance the insurance management system:
- Added new using directives in `PojisteniPojisteneOsobyViewModel.cs`.
- Updated form fields in `AddPolicyToPerson.cshtml` to reflect model changes, including renaming insurance type to policy type.
- Changed model in `NewPerson.cshtml` to `EvidencePojisteniDto.DetailPersonDto` and utilized ASP.NET Core tag helpers for improved validation.
- Added age field in `PersonDetail.cshtml` and updated insurance detail displays.
- Modified model type in `PersonList.cshtml` to `EvidencePojisteniDto.ListItemPersonDto[]` and adjusted table columns and routing parameters.
- Updated policy details display in `PolicyDetail.cshtml` to reflect new property names and improved date and status formatting.

Darkvikis commented on EvidencePojisteni.API/Controllers/ContractController.cs:
```diff
@@ -17,11 +19,12 @@ public class ContractController(ContractService service) : ControllerBase
         var contract = await service.Get(ContractId);
-
-        return contract is null ? NotFound() : Ok(contract);
+        if (contract is null)
```
Brackets

Darkvikis replied:
Also why change it?

Darkvikis commented on EvidencePojisteni.API/Entities/Person.cs:
```diff
@@ -0,0 +1,41 @@
+        FirstName = dto.FirstName;
+        LastName = dto.LastName;
+        Phone = dto.Phone;
+        Age = dto.Age;
```
Arent you missing some stuff?

Darkvikis commented on EvidencePojisteni.API/Entities/Person.cs:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    public void Update(DetailPersonDto dto)
+    {
```
same here

Darkvikis commented on EvidencePojisteni.API/Entities/Person.cs:
```diff
@@ -0,0 +1,41 @@
+    public string LastName { get; set; } = null!;
+    public string Phone { get; set; } = null!;
+    public int Age { get; set; }
+    public Guid ContractId { get; set; }
```
Person can have only one Contract?

Darkvikis commented on EvidencePojisteni.API/Entities/Person.cs:
```diff
@@ -0,0 +1,41 @@
+    public string Phone { get; set; } = null!;
+    public int Age { get; set; }
+    public Guid ContractId { get; set; }
+    public Guid PolicyId { get; set; }
```
Why does it have policy?

Darkvikis commented on EvidencePojisteni.API/Entities/Policy.cs:
```diff
@@ -0,0 +1,32 @@
+    public PolicyType Type { get; set; }
+    public string Name { get; set; } = null!;
+    public Guid ContractId { get; set; }
+    public Guid PersonId { get; set; }
```
Why does it have PersonId

Darkvikis commented on EvidencePojisteni.API/Services/ContractService.cs:
```diff
@@ -1,88 +1,76 @@
 public class ContractService
 {
-    private List<Contract> _contract =
+    private List<ListItemContractDto> _contract =
```
BIG NO

This is you DB. I needs all the info. Where will you get information for Details??? And it should be in entities.

Darkvikis commented on EvidencePojisteni.API/Services/PolicyService.cs:
```diff
@@ -1,69 +1,69 @@
-        {
-            
-        };
+        private readonly List<ListItemPolicyDto> _policy = new();
```
How it should work when is it empty???

Darkvikis commented on EvidencePojisteni.API/Services/PersonService.cs:
```diff
@@ -1,39 +1,47 @@
     public class PersonService
     {
-        private List<Person> _person = new()
+        private List<ListItemPersonDto> _person = new()
```
Empty and Dto???

Darkvikis commented on EvidencePojisteni.API/Services/PolicyService.cs:
```diff
@@ -1,69 +1,69 @@
             // Simulate async operation
             await Task.Delay(100).ConfigureAwait(false);
 
-            return [.. _policy];
```
why?

Darkvikis commented on EvidencePojisteniDto/DetailContractDto.cs:
```diff
@@ -0,0 +1,19 @@
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EvidencePojisteniDto
```
Use this format for name spaces "namespace EvidencePojisteni.API.Entities;"
Jut the format not the whole thing

MartinKonupcik committed:
Refactor namespace and improve DTO handling

Renamed `EvidencePojisteniDto` to `EvidencePojisteniDtos` across the codebase, affecting controllers, services, and DTOs. Streamlined methods in `ContractController.cs` for better readability and consolidated null checks. Updated `ContractService.cs` to utilize `Contract` entities and return `DetailContractDto`. Introduced new DTO classes for contracts, persons, and policies. Modified project files and views to reflect the new structure. Added an `.editorconfig` for consistent coding standards. Updated the solution file to include the new DTO project.

MartinKonupcik committed:
Enhance Person and Policy classes with detail methods

- Added `GetDetail()` method to `Person` class for returning detailed person information.
- Introduced `GetDetail()` method in `Policy` class to provide policy details.
- Updated `PersonService` to replace `_person` with `_personList`, initialized with sample data.
- Modified `Get`, `GetList`, `Create`, `Delete`, and `Update` methods to work with the new structure and return appropriate data types.

MartinKonupcik committed:
Refactor Policy and PolicyService for improved structure

Updated the `Policy` class to include a new `GetDetail` method that returns a `DetailPolicyDto`.

In `PolicyService`, refactored the following:
- Replaced `_policy` with `_policyList` to manage `Policy` objects.
- Modified `Get`, `GetListItems`, `Create`, `Delete`, and `Update` methods to work with the new structure and utilize the `Policy` class methods.
- Adjusted `using` directives for better clarity and organization.

MartinKonupcik committed:
Refactor controllers and update data models

- Refactored methods in `ContractController`, `PersonController`, and `PolicyController` to use ternary operators for cleaner code.
- Updated `Person` class to use lists for `ContractId` and removed `PolicyId`.
- Adjusted related methods and services to accommodate the new data structure.
- Simplified the `Delete` method in `PolicyService`.
- Modified `ListItemPersonDto` to reflect changes in `ContractId` and ensure non-nullable properties for `FirstName` and `LastName`.

MartinKonupcik committed:
Refactor controllers and enhance Policy data models

Updated method names in controllers to align with RESTful conventions. Added new properties to the Policy class and updated related DTOs to include `MaxAmountOfInsuredItems`, `ValidFrom`, and `ValidTo`. Adjusted methods to utilize these new properties in their return values.

MartinKonupcik committed:
Refactor PersonController and update service registrations

- Changed return type of `Get` method in `PersonController` to `DetailPersonDto`.
- Added `PolicyService` as a singleton and registered `PersonService` as scoped in `Program.cs`.
- Updated `ContractId` initialization in `PersonService` from `List<Guid>` to an array.
- Added project reference to `EvidencePojisteni.API` in `EvidencePojisteni.Web.csproj`.
- Switched from `AddControllers` to `AddControllersWithViews` in `Program.cs` for better request handling.
- Enhanced static assets mapping in `Program.cs` with `.WithStaticAssets()`.
- Removed `app.Run()` to restructure application execution.

MartinKonupcik committed:
Remove PersonId from Policy and enhance contract details

This commit removes the `PersonId` property from the `Policy` class, indicating a shift in how policies are associated. Additionally, the `ContractService` class has been updated to include new properties such as `ValidFrom`, `ValidTo`, `Amount`, and `PolicyType` during contract creation. The `PolicyService` class also reflects these changes by adding `MaxAmountOfInsuredItems`, `ValidFrom`, and `ValidTo` properties while removing `PersonId` from policy creation.

MartinKonupcik committed:
Refactor contract handling in person-related classes

Updated the `Person` class to replace `ContractId` with `Contracts`, a list of `Guid`. Adjusted the constructor, `Update`, and `GetListItem` methods accordingly.

Modified `PersonService` to initialize `Contracts` properly and improved method readability. Added `Contracts` property to `DetailPersonDto` for consistency.

Revised `ListItemPersonDto` to replace `ContractId` with the required `Contracts` list, ensuring alignment across related classes.

MartinKonupcik committed:
Add PolicyId to Person and refactor related classes

- Introduced a new `PolicyId` property in the `Person` class.
- Updated `GetListItem` method to include `PolicyId`.
- Cleaned up initialization of `Contract` objects in `ContractService`.
- Modified `Contracts` initialization in `PersonService` to use array-like syntax (note: may need review).
- Adjusted `GetList` method to return an array using spread operator syntax (note: may need review).
- Ensured consistent formatting in `ListItemPersonDto` by removing space in `Contracts` property.

MartinKonupcik committed:
Refactor contract and policy data models

- Updated `Get` method in `ContractController` to return `DetailContractDto`.
- Added `PolicyId` property to `Person` and `DetailPersonDto` classes.
- Included `PolicyId` in `DetailPolicyDto` initialization in `Policy` class.
- Modified `GetListItems` method in `PolicyService` to use spread operator.
- Removed `PersonId` from `ListItemPolicyDto`, changing policy-person relationships.

Darkvikis commented on EvidencePojisteni.API/Services/ContractService.cs:
```diff
@@ -1,88 +1,67 @@
+            Active = true,
+            ValidFrom= DateTime.Now,
+            ValidTo= DateTime.Now.AddYears(1),
+            Amount= 1000m,
```
formatting

Darkvikis commented on EvidencePojisteni.API/Entities/Policy.cs:
```diff
@@ -0,0 +1,52 @@
+{
+    public PolicyType Type { get; set; }
+    public string Name { get; set; } = null!;
+    public Guid ContractId { get; set; }
```
How can I change this?

MartinKonupcik committed:
Refactor policy and person relationship handling

- Removed `PolicyId` from `Person` class and related methods, indicating a shift in how persons relate to policies.
- Updated `Policy` class to replace `ContractId` with a `Contracts` list, allowing multiple contracts per policy.
- Adjusted `PolicyController`, `PersonService`, and `PolicyService` to reflect these changes in their respective logic.
- Modified `ListItemPolicyDto` to align with the new `Contracts` structure.

MartinKonupcik committed:
Refactor policy and person DTOs for consistency

- Updated `Get` method in `PolicyController` to return `DetailPolicyDto`.
- Modified `GetListItem` in `Policy` to include `PolicyId` in the correct position.
- Adjusted `_personList` declaration in `PersonService` for consistency.
- Improved formatting in `Create` method of `PolicyService`.
- Removed `PolicyId` from `DetailPersonDto` and `ListItemPersonDto` to change associations.
- Standardized spacing in `FirstName` and `LastName` properties in `ListItemPersonDto`.

MartinKonupcik committed:
Add Contracts property and refactor services

- Updated `Policy` class to include a new `Contracts` property.
- Changed `PersonService` from scoped to singleton in `Program.cs`.
- Reformatted Swagger UI configuration for clarity.
- Improved formatting consistency in `_contractList` initialization in `ContractService.cs`.
- Added `Contracts` property to `DetailPolicyDto` for better contract representation.
